src/lib.rs

```diff
@@ -172,11 +176,28 @@ impl<T: AsRef<[u8]>> ToBase32<Vec<u5>> for T {
 
 impl Bech32 {
     /// Constructs a `Bech32` struct if the result can be encoded as a bech32 string.
+    ///
+    /// # Errors
+    /// * **MixedCase**: if the HRP contains upper case letters. Encoders must use lower case.
+    /// * **InvalidChar**: if the HRP contains any non-ASCII characters (outside 33..=126)
+    /// * **InvalidLength**: if the HRP is outside 1..83 characters long
+    ///
+    /// # Deviations from standard
+    /// * No length limits are enforced for the data part
```

Comment:
It was much simpler to not enforce this invariant because lightning uses much longer than standard bech32 invoices. But because we need to enforce the same invariants when parsing and constructing that would require two structs `Bech32` and `LenientBech32`. I think it's not worth the added complexity.

Reply:
Agreed.